### PR TITLE
Refine last four account # input for secure send

### PR DESCRIPTION
**Pull request by MuckT**
### Description

A rehashing of #449. With some additional checks for empty strings, backspace functionality and styling changes.

### Other changes

N/A

### Tested

Tested locally on iOS Simulator and Android Emulator.

### Videos

https://user-images.githubusercontent.com/26950305/167179604-38479065-78b7-4183-8b1f-d868a1ad2685.mov

https://user-images.githubusercontent.com/26950305/167181279-d7cca25a-2e56-45e0-929f-fffd1d52488e.mov

### How others should test

1. Attempt to send to phone number registered to multiple accounts e.g. `+12057368924`
2. Select 'Confirm Account Address'
3. Type in any digits in the last four of the account fields and observe:

- Upon typing an input it should navigate to next input automatically.
- Tapping delete should delete the current input and navigate to previous input.
- Empty spaces should not be accepted.
- Placeholders and inputs should remain centered in the input text fields.

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/4116

### Backwards compatibility

Yes

### CODE REVIEW

**Review by jeanregisser — COMMENTED** [review]
Thanks!! 👍 
Since it's a UI change, could you add a video/screenshot to help reviewers/testers?

File: packages/mobile/src/send/ValidateRecipientAccount.tsx [hunk 1]
```diff
@@ -179,9 +183,20 @@ export class ValidateRecipientAccount extends React.Component<Props, State> {
...
+    } else {
+      digitsRefs[index - 1]?.current?.focus()
+    }
     singleDigitInputValueArr[index] = value
     this.setState({ singleDigitInputValueArr })
+    if (singleDigitInputValueArr.filter((entry) => /[a-f0-9]/gi.test(entry)).length === 4) {
+      this.setState({ isFilled: true })
+    } else {
+      this.setState({ isFilled: false })
+    }
```
[the remark this answers is not in the document]
Comment (MuckT) [reply packages/mobile/src/send/ValidateRecipientAccount.tsx]: Fixed in [85035de](https://github.com/valora-inc/wallet/pull/2389/commits/85035de69e15a222e8b2236abee5722c4ab9ddff)!

File: packages/mobile/src/components/SingleDigitInput.tsx [hunk 2]
```diff
@@ -8,20 +8,30 @@ export interface SingleDigitInputProps {
...
+export function SingleDigitInput({
+  inputValue,
+  inputPlaceholder,
+  onInputChange,
+  testID,
+  forwardedRef,
+}: Props) {
   return (
     <TextInput
+      multiline={true}
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply packages/mobile/src/components/SingleDigitInput.tsx:25]: I see, could you add a comment about this?

**Review by jeanregisser — APPROVED** [review]
👍

File: packages/mobile/src/components/SingleDigitInput.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MuckT) [reply packages/mobile/src/components/SingleDigitInput.tsx:28]: Added a comment with the link in https://github.com/valora-inc/wallet/pull/2389/commits/647e921e7752ef9bc3fd992a5b4280387ce4c716!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above issue on Latest Android Internal Release Build V1.32.0 & latest iOS testflight release build V: 1.32.0 and updated the status in Test rails.

**Test rail links:**
**Android 10:** https://valoraapp.testrail.io/index.php?/tests/view/33527
**iOS 15 :** https://valoraapp.testrail.io/index.php?/tests/view/33512
**iOS 14:** https://valoraapp.testrail.io/index.php?/tests/view/33497

**Devices:** Google Pixel XL (10.0) , iPhone 13 mini(15.1.1), iPhone 12(14.7.1)

Thanks..!!

**Pull request merged**